shared/js/content-scripts/element-hiding.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * The purpose of this content script is to hide third-party tracking ads that have been blocked
+ * and appear as blank spaces on websites. In order to distinguish between tracking ads and legitimate
+ * ads, it is necessary to inject this script into all frames.
+ */
+
+var contentScript = {}
+contentScript.frameType = window === window.top ? 'main' : window.parent === window.top ? 'topLevelFrame' : 'nestedFrame'
```

Comment:
After seeing some deeply nested iframes (4+ layers), I decided to limit most of this to top level iframes for performance.